packages/nimble-components/src/table/specs/table-spec.md

```diff
@@ -0,0 +1,242 @@
+
+The other primary risk is simply scope. This is a large, complex component, which can easily grow beyond our initial expectations, due to a variety of unforseen elements, such as missed important features, or misunderstood complexities.
+
+### Prior Art/Examples
+
+SLE Table
+
+![SLE table](./spec-images/sleTable.png)
+
+---
+
+## Design
+
+A user will be able to configure a `nimble-table` component fairly thoroughly through its markup. Including the columns to visualize, how to visualize a column, and in what order. A sketch of the markup should be similar to the following (in Angular):
+
+```html
+<nimble-table [data]="data">
+    <nimble-text-field-column
+        slot="columns"
+        columnId="firstName"
+    ></nimble-text-field-column>
+    <nimble-text-field-column
+        slot="columns"
+        columnId="lastName"
+    ></nimble-text-field-column>
+    <nimble-text-field-column
+        slot="columns"
+        columnId="profession"
+    ></nimble-text-field-column>
+</nimble-table>
+```
+
+_Placeholder for other design details as they are created_
+
+### API
+
+_Component Name_
+
+`nimble-table`
+
+_Feature APIs_
+
+The various APIs/features of the `nimble-table` will be split up amongst several different HLD documents. This section will serve to list them out and link to them as they become available:
+
+-   Data API :
+    -   Define what the interface is for setting/getting data on the table, taking into account any framework-specific binding concerns
+    -   Illustrate how the data is hooked up to the TanStack API
+    -   How is data hierarchy provided and represented in the DOM
+-   Column Definitions
+    -   Define the interface we will provide for the column providers/components (i.e., width, sorting, allowSort, allowGrouping, etc...)
+    -   Define the base implementation for the column providers that other column providers can extend.
+    -   List the set of column providers that Nimble will provide and provide their respective APIs where unique (e.g., formatter for DateTime column)
+    -   Define the table-level APIs for setting the columns
+        -   Slot API on table?
+        -   Alternative/accompanying programmatic API?
+-   Headers
+    -   Define the anatomy of headers in the table DOM
+        -   Require specific component type?
+        -   What is the component to use for interaction? Outline Button? Ghost button?
+        -   What and where are the interactive mechanisms/indicators? Sort arrow, etc..
+-   Row Selection
+    -   Define the anatomy of row selection in the table DOM
+        -   Indeterminate checkbox at the far left of each row?
+        -   Selected row CSS
+    -   Define events raised when row selection changes/occurs
+    -   Define table-level row-selection API
+-   Grouping
```

Comment:
Maybe add a bullet for how grouping works with hierarchy.

Reply:
Done.